pkg/api/serialization_test.go

```diff
@@ -131,6 +132,22 @@ func roundTripSame(t *testing.T, group testapi.TestGroup, item runtime.Object, e
 	}
 }
 
+func TestSetControllerConversion(t *testing.T) {
+	rs := &extensions.ReplicaSet{}
+	gv := testapi.Extensions
+	fuzzInternalObject(t, gv.InternalGroupVersion(), rs, rand.Int63())
+
+	data, err := runtime.Encode(gv.Codec(), rs)
+	if err != nil {
+		t.Errorf("unexpected encoding error: %v", err)
+	}
+
+	rc := &v1.ReplicationController{}
+	if err := runtime.DecodeInto(testapi.Default.Codec(), data, rc); err != nil {
```

Comment:
Hm. The codec won't do cross-group conversions by default. Unfortunately, you'll have to do something like call https://github.com/kubernetes/kubernetes/blob/master/pkg/runtime/serializer/versioning/versioning.go#L31. I think ideally we'd somehow represent this information in the scheme.

...actually now that I look more closely, you're using the testapi.Default codec, but I'm not sure if that one would have both objects in any case. Try testapi.Extensions?

The thing you're trying to add (single object alternate conversion via separate group) is not an expected use case and that means there's going to be issues like this.